### PR TITLE
Pin macos-15 build runner

### PR DESCRIPTION
**Pull request by jamescrosswell**
Resolves https://github.com/getsentry/sentry-dotnet/issues/3752

#skip-changelog

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -63,31 +63,19 @@ jobs:
...
-        if: matrix.os == 'macos-latest'
-        run: |
-          df -hI /dev/disk3s1s1
-          sudo rm -rf /Applications/Xcode_14.3.1.app
-          sudo rm -rf /Applications/Xcode_15.0.1.app
-          sudo rm -rf /Applications/Xcode_15.1.app
-          sudo rm -rf /Applications/Xcode_15.2.app
-          sudo rm -rf /Applications/Xcode_15.3.app
-          df -hI /dev/disk3s1s1
-
```
Comment (bitsandfoxes) [inline .github/workflows/build.yml:90]: 👍

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -15,11 +15,11 @@ env:
 jobs:
   build-sentry-native:
     name: sentry-native (${{ matrix.os }})
-    runs-on: ${{ matrix.os }}-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu, windows, macos]
+        os: [ubuntu-latest, windows-latest, macos-15]
```
Comment (bitsandfoxes) [inline .github/workflows/build.yml:22]: Since we're pinning it here, can we add a comment as to why?

**Review by bitsandfoxes — APPROVED** [review]
The ever changing flow of pin/unpin...

**Comment by jamescrosswell** [comment]
We're getting this error on the macos-15 runner:
```
Error: /Users/runner/.dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.145/targets/Microsoft.Android.Sdk.Tooling.targets(31,5): error XA5205: Cannot find `aapt`. Please install the Android SDK Build-Tools package with the `/Users/runner/Library/Android/sdk/tools/android` program. [/Users/runner/work/sentry-dotnet/sentry-dotnet/samples/Sentry.Samples.Maui/Sentry.Samples.Maui.csproj::TargetFramework=net8.0-android]
```

Looks like that runner doesn't (yet) include the android build tools. Workaround described here:
https://github.com/actions/runner-images/issues/10814#issuecomment-2442435678

```yml
      # https://github.com/actions/runner-images/issues/10814
      - name: Workaround build-tools issue
        run: | 
          curl https://dl.google.com/android/repository/build-tools_r35_macosx.zip > $ANDROID_HOME/build-tools_r35_macosx.zip
          cd $ANDROID_HOME
          mkdir build-tools
          unzip build-tools_r35_macosx.zip
          mv android-15 build-tools/35.0.0
```

**Comment by bruno-garcia** [comment]
>   [+] Ensure System.Reflection.Metadata is not missing 

[Is this a integration](https://github.com/getsentry/sentry-dotnet/actions/runs/11845343488/job/33012837228?pr=3754) test flake @vaind?

image
image

I actually don't understand the failure

**Comment by jamescrosswell** [comment]
> [Is this a integration](https://github.com/getsentry/sentry-dotnet/actions/runs/11845343488/job/33012837228?pr=3754) test flake @vaind?

It's not the reason the integration test failed... that was because of this:
```
    /Users/runner/.dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.145/tools/Xamarin.Android.Tooling.targets(100,5): error XA5207: Could not find android.jar for API level 34. This means the Android SDK platform for API level 34 is not installed; it was expected to be in `/Users/runner/Library/Android/sdk/platforms/android-34/android.jar`. [/maui-app/maui-app.csproj::TargetFramework=net8.0-android]
    /Users/runner/.dotnet/packs/Microsoft.Android.Sdk.Darwin/34.0.145/tools/Xamarin.Android.Tooling.targets(100,5): error XA5207: You can install the missing API level by running `dotnet build -t:InstallAndroidDependencies -f net8.0-android "-p:AndroidSdkDirectory=/Users/runner/Library/Android/sdk"`, or change the project to target an API version that is installed. [/maui-app/maui-app.csproj::TargetFramework=net8.0-android]
```

The Android SDKs obviously aren't included on the macos-15 runner, so we need to install these ourselves. I already added that for the build step:
https://github.com/getsentry/sentry-dotnet/blob/e6e6424b848627ebadecd1c97f49979f0e5a5feb/.github/workflows/build.yml#L126-L130

With any luck, this does the same for the integration tests:
https://github.com/getsentry/sentry-dotnet/blob/40b9757db41908b038cdcecb18884cead330263d/integration-test/cli.Tests.ps1#L145-L152

It's kind of painful to work through all these issues since they only happen on CI... and now we're getting to the last ones (in the integration tests) there's a minimum 60-90 minute wait after each change to see if it worked.

**Pull request merged**